air/src/proof/context.rs

```diff
@@ -18,18 +18,24 @@ pub struct Context {
     trace_info: TraceInfo,
     field_modulus_bytes: Vec<u8>,
     options: ProofOptions,
+    total_num_of_constraints: usize,
 }
 
 impl Context {
     // CONSTRUCTOR
     // --------------------------------------------------------------------------------------------
-    /// Creates a new context for a computation described by the specified field, trace info, and
-    /// proof options.
+    /// Creates a new context for a computation described by the specified field, trace info, proof
+    /// options, and total number of constraints.
     ///
     /// # Panics
-    /// Panics if either trace length or the LDE domain size implied by the trace length and the
-    /// blowup factor is greater then [u32::MAX].
-    pub fn new<B: StarkField>(trace_info: TraceInfo, options: ProofOptions) -> Self {
+    /// - If either trace length or the LDE domain size implied by the trace length and the
+    ///   blowup factor is greater then [u32::MAX].
+    /// - If the number of constraints is greater than [u32::MAX].
+    pub fn new<B: StarkField>(
+        trace_info: TraceInfo,
+        options: ProofOptions,
+        total_num_of_constraints: usize,
+    ) -> Self {
```

Comment:
nit: I would rename into just `num_constraints` here and in other places.

Also, we should probably make sure that the number of constraints is greater than 0.

Reply:
updated